From 17b5894905339ba11cb7a75b6134a9dfa3216213 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 7 Dec 2017 19:51:07 +0900
Subject: [PATCH 386/909] media: i2c: adv7604: Enable decoder control register

commit c3d133c37918c43b30bf485f9ec37e81482aa1ff from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Since bit for enabling the color conversion function
of the decoder was not set up, this patch enables it.
Also, because the setting range value of HUE was different
from the manual, it was fixed.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/adv7604.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/adv7604.c b/drivers/media/i2c/adv7604.c
index 22843ebcc8eb..dcf23a880689 100644
--- a/drivers/media/i2c/adv7604.c
+++ b/drivers/media/i2c/adv7604.c
@@ -2,6 +2,7 @@
 /*
  * adv7604 - Analog Devices ADV7604 video decoder driver
  *
+ * Copyright (C) 2017 Renesas Electronics Corp.
  * Copyright 2012 Cisco Systems, Inc. and/or its affiliates. All rights reserved.
  *
  */
@@ -70,6 +71,8 @@ MODULE_LICENSE("GPL");
 
 #define ADV76XX_OP_SWAP_CB_CR				(1 << 0)
 
+#define ADV7612_CP_VID_ADJ_ENABLE			(1 << 7)
+
 #define ADV76XX_MAX_ADDRS (3)
 
 enum adv76xx_type {
@@ -1192,6 +1195,15 @@ static int adv76xx_s_ctrl(struct v4l2_ctrl *ctrl)
 		&container_of(ctrl->handler, struct adv76xx_state, hdl)->sd;
 
 	struct adv76xx_state *state = to_state(sd);
+	int ret;
+	u8 val;
+
+	/* Enable video adjustment first */
+	val = cp_read(sd, 0x3e);
+	val |= ADV7612_CP_VID_ADJ_ENABLE;
+	ret = cp_write(sd, 0x3e, val);
+	if (ret < 0)
+		return ret;
 
 	switch (ctrl->id) {
 	case V4L2_CID_BRIGHTNESS:
@@ -3450,7 +3462,7 @@ static int adv76xx_probe(struct i2c_client *client,
 	v4l2_ctrl_new_std(hdl, &adv76xx_ctrl_ops,
 			V4L2_CID_SATURATION, 0, 255, 1, 128);
 	v4l2_ctrl_new_std(hdl, &adv76xx_ctrl_ops,
-			V4L2_CID_HUE, 0, 128, 1, 0);
+			V4L2_CID_HUE, 0, 255, 1, 0);
 	ctrl = v4l2_ctrl_new_std_menu(hdl, &adv76xx_ctrl_ops,
 			V4L2_CID_DV_RX_IT_CONTENT_TYPE, V4L2_DV_IT_CONTENT_TYPE_NO_ITC,
 			0, V4L2_DV_IT_CONTENT_TYPE_NO_ITC);
-- 
2.17.1

